core/orchestrator.go

```diff
@@ -856,35 +873,45 @@ func (rtm *RemoteTranscoderManager) selectTranscoder() *RemoteTranscoder {
 	}
 
 	for checkTranscoders(rtm) {
+		currentTranscoder, sessionExists := rtm.streamSessions[sessionId]
+		if sessionExists {
+			return currentTranscoder, nil
+		}
+
 		last := len(rtm.remoteTranscoders) - 1
-		currentTranscoder := rtm.remoteTranscoders[last]
+		currentTranscoder = rtm.remoteTranscoders[last]
+
 		if _, ok := rtm.liveTranscoders[currentTranscoder.stream]; !ok {
 			// transcoder does not exist in table; remove and retry
 			rtm.remoteTranscoders = rtm.remoteTranscoders[:last]
```

Comment:
Doesn't look like we're currently removing a transcoder from this slice if the transcoder was selected for an existing stream session (as opposed to just being the last transcoder in the slice). We'll need to figure out a way to do this given that in this scenario we can't just remove the last transcoder from the slice since the transcoder that was selected may not necessarily be the last in the slice.

cc @kyriediculous to advise here.

Reply:
The only possibility I see is looping through the slice if last != currentTranscoder 

We are already resorting this list constantly anyway so more iterative behaviour doesn't seem like it would hurt too much. Not that it's optimal and this entire system of tracking a map of connections alongside an array of transcoders is very weird and should likely be redesigned at some point to be a unified and single data structure tracking remote transcoders.

Reply:
Sounds reasonable for now.

Reply:
@reubenr0d perhaps we can create a functional helper for this something like 

```go
func removeFromRemoteTranscoders(rt *RemoteTranscoder, remoteTranscoders []*RemoteTranscoder) []*RemoteTranscoder {
	lastIndex := len(remoteTranscoders) - 1
	last := remoteTranscoders[lastIndex]
	if rt == last {
		return remoteTranscoders[:lastIndex]
	}

	newRemoteTs := make([]*RemoteTranscoder, 0)
	for i, t := range remoteTranscoders {
		if t == rt {
			if i == 0 {
				return remoteTranscoders[1:]
			}
			newRemoteTs = remoteTranscoders[i-1 : i]
			newRemoteTs = append(newRemoteTs, remoteTranscoders[i+1:]...)
			break
		}
	}
	return newRemoteTs
}
```

And then replace L886 with a call to this function